core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2831,10 +2831,16 @@ private[spark] class DAGScheduler(
     failedStage.latestInfo.completionTime = Some(clock.getTimeMillis())
     updateStageInfoForPushBasedShuffle(failedStage)
     for (job <- dependentJobs) {
-      failJobAndIndependentStages(
-        job,
+      val finalException = exception.collect {
+        // If the error is well defined (has an error class and is not internal error), we treat
+        // it as user-facing, and expose this error to the end users directly.
```

Comment:
This includes more exception types than just those that match isRuntimeUserError() right? I think we might want a different name somewhere, people might get user-facing error and user error mixed up?